### DISCUSSION TITLE
How would you approach making any filesystem changes on system appearance change

### DISCUSSION THREAD

**Discussion by skoch13** [wezterm/wezterm#5121] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I am using `wezterm.gui.get_appearance()` to react to system appearance changes. However, there are other places that also need to be updated, such as:

-  Environment variables: `BAT_THEME`, `FZF_DEFAULT_OPTS`, etc.
-  `.gitconfig`'s `delta` settings.

What would be the suggested approach within wezterm's capabilities, or should it be addressed elsewhere? Thank you in advance.

**Comment by wez** [comment, marked as answer]
When the appearance changes, the config is reloaded and the entire file is re-evaluated.
You could update any config files in the main scope if you wish, but I'd suggest that you use something like https://wezfurlong.org/wezterm/config/lua/wezterm/GLOBAL.html to remember the last appearance value that you saw so that you don't trigger a lot of work for other config re-evaluation scenarios; see the first "info" block on https://wezfurlong.org/wezterm/config/files.html for more information.

re: updating environment variables: it is not possible on any OS for a process to change the environment of a target process after it has been spawned. If your goal is to detect an appearance change in your shell then I would recommend that on the wezterm side you write out a file with the appearance info, then in your shell have a precmd hook or something that will detect changes to that file and update its own environment.

**Reply by skoch13** [reply]
thanks!
